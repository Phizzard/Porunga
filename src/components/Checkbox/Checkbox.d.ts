import { Rounded, Palette } from '../../types'
/**
 * Checkbox properties
 */
export interface CheckboxProps {
  /** Content inside the checkbox element */
  onChange?(isChecked: boolean): any
  id?: string
  name?: string
  checked?: boolean
  disabled?: boolean
  className?: string
  primaryColor?: Palette
  label: string
  rounded?: Rounded
}
/**
 * Checkbox
 */
export declare const Checkbox: ({
  onChange,
  id,
  name,
  checked,
  disabled,
  className,
  primaryColor,
  label,
  rounded,
}: CheckboxProps) => JSX.Element
export default Checkbox
