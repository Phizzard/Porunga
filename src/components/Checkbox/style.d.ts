/// <reference types="react" />
import { SerializedStyles } from '@emotion/core'
interface CheckboxProps {
  primaryColor?: string
  disabled?: boolean
}
interface CustomCheckboxProps {
  primaryColor?: string
  isChecked: boolean
  rounded?: SerializedStyles
}
export declare const Container: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').LabelHTMLAttributes<HTMLLabelElement>,
    HTMLLabelElement
  >,
  CheckboxProps,
  object
>
export declare const Input: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  >,
  CheckboxProps,
  object
>
export declare const CustomCheckbox: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  >,
  CustomCheckboxProps,
  object
>
export {}
