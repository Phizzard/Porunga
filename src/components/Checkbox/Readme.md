Default Checkbox

```jsx
<Checkbox>Default Checkbox</Checkbox>
```
