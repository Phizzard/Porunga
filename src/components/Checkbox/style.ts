import styled from '@emotion/styled'

interface CheckboxProps {
  backgroundColor?: string
}

interface CustomCheckboxProps {
  backgroundColor?: string
  isChecked: boolean
}

export const Container = styled.label<CheckboxProps>`
  position: relative;
  color: ${p => p.backgroundColor};
  padding: 0.6rem 1.1rem;
  font-size: 1.1rem;
  font-weight: 500;
  border-radius: 24px;
  cursor: pointer;
  background-color: #f7fafc;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  transition: 0.2s;

  :hover,
  :focus {
    background-color: ${p => p.backgroundColor};
    color: #fff;

    > div {
      background-color: #fff;
      color: ${p => p.backgroundColor};
    }
  }
`

export const Input = styled.input<CheckboxProps>`
  position: absolute;
  margin: 0;
  padding: 0;
  opacity: 0;

  :focus + div {
    background-color: #fff;
    color: ${p => p.backgroundColor};
  }
`

export const CustomCheckbox = styled.div<CustomCheckboxProps>`
  position: relative;
  display: inline-block;
  vertical-align: middle;
  background-color: ${p => p.backgroundColor};
  color: #fff;
  margin: -3px 5px 0 0;
  padding: 0;
  font-size: 1.1rem;
  width: 20px;
  height: 20px;
  border-radius: 5px;

  &:after {
    content: 'L';
    position: absolute;
    top: 0;
    left: 9px;
    transform: scaleX(-1) rotate(-45deg);
    width: 10px;
    height: 10px;
    visibility: ${p => (p.isChecked ? 'visible' : 'hidden')};
  }
`
