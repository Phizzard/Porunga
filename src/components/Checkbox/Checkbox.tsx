import React, { useState } from 'react'
import { useTheme } from 'emotion-theming'
import cx from 'classnames'

import { ThemeConfig } from '../../types'
import { get } from '../../utils/get'
import { Container, Input, CustomCheckbox } from './style'

/**
 * Checkbox properties
 */
export interface CheckboxProps {
  /** Content inside the checkbox element */
  onChange?(isChecked: boolean, event: object): any
  id?: string
  name?: string
  disabled?: boolean
  className?: string
  backgroundColor?: string
  children?: React.ReactNode
}

/**
 * Checkbox
 */
export const Checkbox = ({
  onChange,
  id,
  name,
  disabled,
  className,
  backgroundColor,
  children,
}: CheckboxProps) => {
  const [isChecked, setIsChecked] = useState(false)
  const theme: ThemeConfig = useTheme()

  const checkboxProps = {
    backgroundColor: backgroundColor || get(theme, 'root', 'colors', 'brand'),
  }

  const handleChange = (event: object) => {
    const newState = !isChecked
    setIsChecked(newState)
    if (onChange) onChange(newState, event)
  }

  return (
    <Container
      id={id}
      className={cx(className, { disabled })}
      {...checkboxProps}
    >
      <Input
        type="checkbox"
        name={name}
        {...checkboxProps}
        checked={isChecked}
        onChange={handleChange}
        disabled={disabled}
      />
      <CustomCheckbox {...checkboxProps} isChecked={isChecked} /> {children}
    </Container>
  )
}

export default Checkbox
