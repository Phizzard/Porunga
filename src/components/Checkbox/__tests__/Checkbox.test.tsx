import React from 'react'
import renderer from 'react-test-renderer'
import Checkbox from '../Checkbox'

it('renders correctly', () => {
  const component = renderer.create(<Checkbox label="test" />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
