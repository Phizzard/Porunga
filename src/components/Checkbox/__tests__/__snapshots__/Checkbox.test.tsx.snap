// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className=" css-1mk237q"
>
  <input
    checked={false}
    className="css-11c0hjf"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="css-k8kyit"
  />
   
</label>
`;
